spec/models/metric/ci_mixin/capture_spec.rb

```diff
@@ -36,6 +36,16 @@ def expected_stats_period_end
     parse_datetime('2013-08-28T12:41:40Z')
   end
 
+  context "#perf_capture_queue" do
+    it "splits up long perf_capture durations for old last_perf_capture_on" do
+      # there should be 10 days + a partial day, or 11 queue items
+      original_queue_count = MiqQueue.count
+      @vm.last_perf_capture_on = 10.days.ago - 5.hours - 23.minutes
+      @vm.perf_capture_queue("realtime")
+      expect(MiqQueue.count - original_queue_count).to eq 11
```

Comment:
It might be good to check some candidate time values and also to see if they are in reverse.  You can use TimeCop.freeze if you need to ensure specific time values.